pytensor/scalar/basic.py

```diff
@@ -1533,6 +1533,56 @@ def c_code_cache_version(self):
 isinf = IsInf()
 
 
+class IsPosInf(FixedLogicalComparison):
+    nfunc_spec = ("isposinf", 1, 1)
+
+    def impl(self, x):
+        return np.isposinf(x)
+
+    def c_code(self, node, name, inputs, outputs, sub):
+        (x,) = inputs
+        (z,) = outputs
+        if node.inputs[0].type in complex_types:
+            raise NotImplementedError()
+        # Discrete type can never be posinf
+        if node.inputs[0].type in discrete_types:
+            return f"{z} = false;"
+
+        return f"{z} = isinf({x}) && !signbit({x});"
+
+    def c_code_cache_version(self):
+        scalarop_version = super().c_code_cache_version()
+        return (*scalarop_version, 4)
+
+
+isposinf = IsPosInf()
+
+
+class IsNegInf(FixedLogicalComparison):
```

Comment:
Do we need these custom Ops, or can we just use helper functions like `nan_to_num` such as `pt.eq(x, -np.inf)`?

In general we want to have as little Ops as possible, and just reuse what we already have.

Reply:
I just referred to [this](https://github.com/pymc-devs/pytensor/pull/540#discussion_r1421410406) conversation and implemented the two ops. This can be done even without them. Should I omit these in the next commit?

Reply:
Yup, no new Ops is always the default and preferred strategy